### PR TITLE
feat(config): support task command shorthands

### LINKED ISSUE #381: Support task string shorthand and command string[] shorthand
### Summary

Implement the two config simplifications discussed in Discord:

1. Allow task shorthand:`"some-task": "... command"` (equivalent to `"some-task": { "command": "... command" }`)
2. Allow `command` to accept `string[]` as a shorthand for `command1 && comand2 && ...`.

### Notes

- The task string shorthand should keep the same default caching behavior as the object form (`run.cache` still overrides, default remains `true`).
- Docs should make sequencing semantics explicit.
- These two shorthands can be combined: `"some-task": ["cmd1", "cmd2"]`
- Verifiable by plan snapshot tests. (no need for e2e tests)

### PR DESCRIPTION
**Pull request by jong-kyung**
## Summary

Closes #381.

Adds task command shorthands:

- `"build": "cmd"`
- `"build": ["cmd1", "cmd2", ...]`
- `{ "command": ["cmd1", "cmd2", ...] }`

Arrays reuse the existing `&&` planning path, so cache behavior, `dependsOn`, and task options stay consistent. Empty arrays and empty/whitespace-only entries are rejected.

### CODE REVIEW

**Comment by jong-kyung** [comment]
@codex review

File: crates/vite_task_graph/src/config/user.rs [hunk 1]
```diff
@@ -193,20 +193,116 @@ impl Default for UserTaskOptions {
...
+// TS derive macro generates code using std types that clippy disallows; skip derive during linting
+#[cfg_attr(all(test, not(clippy)), derive(TS), ts(type = "string | Array<string>"))]
+pub enum TaskCommand {
+    /// A raw command string.
+    String(Str),
+    /// Command snippets to join with ` && `.
+    Array(Vec<Str>),
+}
+
+impl<'de> Deserialize<'de> for TaskCommand {
```
Comment (wan9chi) [inline crates/vite_task_graph/src/config/user.rs]: The manual `Deserialize` impl seems unnecessary. Consider use [`#[serde(untagged)]`](https://serde.rs/enum-representations.html#untagged) on `TaskCommand`.

File: crates/vite_task_graph/src/config/user.rs [hunk 2]
```diff
@@ -193,20 +193,116 @@ impl Default for UserTaskOptions {
...
+impl TaskCommand {
+    #[must_use]
+    pub fn into_command_string(self) -> Str {
+        match self {
+            Self::String(command) => command,
+            Self::Array(commands) => {
+                let mut commands = commands.into_iter();
+                let mut command = commands.next().expect("command arrays are non-empty");
+                for item in commands {
+                    command.push_str(" && ");
```
Comment (wan9chi) [inline crates/vite_task_graph/src/config/user.rs]: This data flow can't handle cases like `command: ["a '", "b '"]` (it'd generate `a ' && ' b`).

Instead we should:

1.  `string` command ->  `string[]` command with a single element.
2. For every item in `string[]` command, parse it into array of `TaskParsedCommand`
3. Concat all `TaskParsedCommand`

Comment (wan9chi) [reply crates/vite_task_graph/src/config/user.rs]: This file is only for data definitions.
The logic of converting and parsing `command` should be in `vite_task_plan`.

File: crates/vite_task_graph/src/config/user.rs [hunk 3]
```diff
@@ -193,20 +193,116 @@ impl Default for UserTaskOptions {
...
+
+impl UserTaskDefinition {
+    #[must_use]
+    pub fn into_config(self) -> UserTaskConfig {
+        match self {
+            Self::Config(config) => config,
+            Self::Command(command) => {
+                UserTaskConfig { command, options: UserTaskOptions::default() }
+            }
+        }
```
Comment (wan9chi) [inline crates/vite_task_graph/src/config/user.rs]: Move this to `vite_task_plan`, too.

File: crates/vite_task_graph/src/config/user.rs [hunk 4]
```diff
@@ -193,20 +193,116 @@ impl Default for UserTaskOptions {
     }
 }
 
+/// Task command: a string, or command snippets joined with ` && `.
+///
+/// Arrays are not argv-style and element boundaries are not preserved after joining.
```
Comment (wan9chi) [inline crates/vite_task_graph/src/config/user.rs]: Doc comments in this file will be generated in `run-config.ts`, and be seen by users when they create tasks. Let's make them easier to understand without implementation details.

File: crates/vite_task_plan/tests/plan_snapshots/fixtures/task_command_shorthands/vite-task.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    "object_array_cache_false": {
+      "command": ["vtt print-file package.json", "vtt print-file vite-task.json", "vtt print-file package.json"],
+      "cache": false
+    },
+    "object_array_depends_on": {
+      "command": ["vtt print-file package.json", "vtt print-file vite-task.json", "vtt print-file package.json"],
+      "dependsOn": ["string_shorthand"]
+    }
+  }
+}
```
Comment (wan9chi) [inline crates/vite_task_plan/tests/plan_snapshots/fixtures/task_command_shorthands/vite-task.json:29]: More things to test:

- nested `vt run ...` inside the command array should be expanded in the plan
- Combine `&&` and array (`["foo", "bar && baz"]`)

File: crates/vite_task_plan/tests/plan_snapshots/fixtures/task_command_shorthands_cache_disabled/vite-task.json
Comment (wan9chi) [inline crates/vite_task_plan/tests/plan_snapshots/fixtures/task_command_shorthands_cache_disabled/vite-task.json:1]: Do we have special logic on cache-disabled tasks? If not, then `task_command_shorthands_cache_disabled` is not necessary.

Comment (jong-kyung) [reply crates/vite_task_plan/tests/plan_snapshots/fixtures/task_command_shorthands_cache_disabled/vite-task.json:1]: Agreed. I’ll remove this unnecessary test.

File: crates/vite_task_plan/tests/plan_snapshots/fixtures/task_command_shorthands/vite-task.json (hunk 5 above)
Comment (jong-kyung) [reply crates/vite_task_plan/tests/plan_snapshots/fixtures/task_command_shorthands/vite-task.json:29]: Thanks for the suggestions. I’ll add more tests for these cases.

There are definitely several edge cases I hadn’t considered.

**Comment by jong-kyung** [comment]
@codex review

**Comment by jong-kyung** [comment]
@codex review

**Comment by jong-kyung** [comment]
@codex review

**Comment by jong-kyung** [comment]
@branchseer I've addressed all of the suggested changes from your review and incorporated the feedback from Codex as well. Could you please review it again when you have a chance?

**Comment by jong-kyung** [comment]
@codex review

File: crates/vite_task_graph/src/config/mod.rs [hunk 6]
```diff
@@ -28,10 +28,10 @@ use crate::config::user::UserTaskOptions;
 /// `depends_on` is not included here because it's represented by the edges of the task graph.
 #[derive(Debug, Serialize)]
 pub struct ResolvedTaskConfig {
-    /// The command to run for this task, as a raw string.
+    /// The command or commands to run for this task.
     ///
-    /// The command may contain environment variables that need to be expanded later.
-    pub command: Str,
+    /// Commands may contain environment variables that need to be expanded later.
+    pub command: TaskCommand,
```
Comment (wan9chi) [inline crates/vite_task_graph/src/config/mod.rs]:
```suggestion
    /// The commands to run for this task.
   ///
    /// Commands may contain environment variables that need to be expanded later.
    pub commands: Arc<[Str]>,
```

`TaskCommand::String` can be resolved to an array with a single element in `ResolvedTaskConfig` (Treat `command: "hello"` as `command: ["hello"]`), so that `vite_task_plan` wouldn't have to handle the two `TaskCommand` variants.

File: crates/vite_task_plan/src/plan.rs [hunk 7]
```diff
@@ -80,6 +81,54 @@ fn effective_cache_config(
...
+            if commands.iter().any(|command| command.as_str().trim().is_empty()) {
+                return Err(Error::InvalidTaskCommand(
+                    "command array entries must not be empty".into(),
+                ));
+            }
+
+            let mut source = Str::default();
+            for (index, command) in commands.iter().enumerate() {
+                if index > 0 {
+                    source.push_str(" && ");
```
Comment (wan9chi) [inline crates/vite_task_plan/src/plan.rs]: Please don't re-join to string and then parse. It can't handle `command: ["foo '", "' bar"]`.

We should parse elements in `commands` individually.

There will be two level of command items. We should add a `command_item_index: usize` to `ExecutionCacheKey::UserTask` alongside `and_item_index`.

File: crates/vite_task_plan/src/plan.rs [hunk 8]
```diff
@@ -128,264 +176,271 @@ async fn plan_task_as_execution_node(
...
+                };
 
-        let mut script = Str::from(command_str);
-        for arg in context.extra_args().iter() {
-            script.push(' ');
-            script.push_str(shell_escape::escape(arg.as_str().into()).as_ref());
-        }
+                let execution_item_display = ExecutionItemDisplay {
+                    command: script.clone(),
+                    and_item_index: if and_item_count > 1 { Some(index) } else { None },
```
Comment (wan9chi) [inline crates/vite_task_plan/src/plan.rs]: `ExecutionItemDisplay.and_item_index` is actually dead code. Sorry. I'm removing it in https://github.com/voidzero-dev/vite-task/pull/394.

`ExecutionCacheKey::UserTask.and_item_index` is still useful and we still need a `command_item_index` there.

**Comment by jong-kyung** [comment]
@codex review

**Comment by jong-kyung** [comment]
@branchseer I’ve addressed the suggested changes. Could you please review it again? 🙇‍♂️

**Comment by wan9chi** [comment]
Thanks!
